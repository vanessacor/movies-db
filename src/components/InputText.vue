<template>
	<div class="input-wrapper">
		<label class="input-label">{{ label }}</label>
		<input
			class="input"
			v-bind="$attrs"
			v-bind:value="value"
			v-on:input="$emit('input', $event.target.value)"
		/>
	</div>
</template>

<script>
	export default {
		name: "inputText",
		inheritAttrs: false,
		props: ["label", "value"],
	};
</script>

<style lang="scss">
	.input-wrapper {
		display: flex;
		flex-direction: column;
	}
	.input {
		padding: 10px;
		width: 100%;
		font-size: 16px;
		box-shadow: none;
		outline: none;
		border: none;
		background-color: transparent;
		border-bottom: 2px solid $base-color-light;
		color: $base-color-light;
	}
	.input-label {
		margin-bottom: 5px;
		
	}
</style>