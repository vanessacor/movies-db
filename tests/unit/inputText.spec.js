import { mount } from "@vue/test-utils";
import InputText from "@/components/InputText";

describe("test InputText.vue", () => {
  let wrapper = null;
  const value = "Vanessa trip";
  const label = "Insert the title here";
  beforeEach(() => {
    wrapper = mount(InputText, {
      propsData: { value, label },
    });
  });
  afterEach(() => {
    wrapper.destroy();
  });
  it("renders props when passed", () => {
    expect(wrapper.vm.value).toMatch(value);
    expect(wrapper.vm.label).toMatch(label);
  });

  it("emits an event", () => {
    wrapper.vm.$emit("input");
    expect(wrapper.emitted().input).toBeTruthy();
  });

  it("match the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
