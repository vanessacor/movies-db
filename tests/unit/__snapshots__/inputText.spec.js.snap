// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test InputText.vue match the snapshot 1`] = `<div class="input-wrapper"><label class="input-label">Insert the title here</label> <input class="input"></div>`;
